src/sql/workbench/parts/dataExplorer/browser/dataExplorer.contribution.ts

```diff
@@ -88,3 +94,95 @@ configurationRegistry.registerConfiguration({
 		}
 	}
 });
+
+export interface IDataExplorerActionContribution {
+	category: string;
+	commandId: string;
+	label: string;
+	icon?: {
+		light: string;
+		dark: string;
+	};
+	isPrimary: boolean;
+}
+
+const DataExplorerActionSchema: IJSONSchema = {
+	type: 'object',
+	properties: {
+		category: {
+			type: 'string',
+			description: localize('azdata.extension.contributes.dataExplorer.action.category', "Category of the action used for grouping the actions")
+		},
+		commandId: {
+			type: 'string',
+			description: localize('azdata.extension.contributes.dataExplorer.action.commandId', "Id of the command that will be executed")
+		},
+		label: {
+			type: 'string',
+			description: localize('azdata.extension.contributes.dataExplorer.action.label', "Display name of the action")
+		},
+		icon: {
+			type: 'object',
+			description: localize('azdata.extension.contributes.dataExplorer.action.icon', "Icon for the action")
+		},
+		isPrimary: {
+			type: 'boolean',
+			description: localize('azdata.extension.contributes.dataExplorer.action.isPrimary', "Indicates whether the action is a primary action")
+		}
+	}
+};
+
+ExtensionsRegistry.registerExtensionPoint<IDataExplorerActionContribution | IDataExplorerActionContribution[]>({ extensionPoint: 'dataExplorer.actions', jsonSchema: DataExplorerActionSchema }).setHandler(extensions => {
```

Comment:
Whats the purpose of this? Why do we have a custom entry for this?

Reply:
is there already a way to contribute actions for data explorer?

Reply:
You can just use the standard menu contribution 'dataExplorer'. There is a problem there, but this isn't the right solution.

Reply:
@anthonydresser , I'd love to know it if we already have such feature.

Reply:
What are you trying to do? Just add menu contributions? Why isn't the data explorer menu contribution enough for that?

Reply:
![image](https://user-images.githubusercontent.com/13777222/57646458-7a994800-7575-11e9-998a-7e948df13fee.png)

this is what I showed in the review meeting, I might end up only using the secondary actions here. I don't think we have support for this right now. could you please confirm? @anthonydresser

Reply:
As mentioned earlier, I would suggest using existing Menus contribution as suggested by Anthony but supporting this new `isPrimary` option to hide in a `...`. That could be done by extending the schema and just having the places we pick up & implement the contribution on the SQL side read & use those properties.

Reply:
thanks for the feedback, figured out an easier way to contribute the menu items, I introduced a new type instead of using the dataExplorer/context to avoid affecting the context menus.